src/pudl/transform/eia923.py

```diff
@@ -1223,3 +1223,96 @@ def _core_eia923__fuel_receipts_costs(
+
+    Returns:
+        Cleaned ``_core_eia923__fgd_operation_maintenance`` dataframe ready for harvesting.
+    """
+    fgd_df = raw_eia923__fgd_operation_maintenance
+
+    # Drop duplicate and empty rows to ensure the primary key is unique.
+    fgd_df = fgd_df.dropna(subset=["plant_id_eia", "so2_control_id_eia"])
+    fgd_df = fgd_df.drop_duplicates()
+
+    # Replace the EIA923 NA value ('.') with a real NA value.
+    fgd_df = pudl.helpers.fix_eia_na(fgd_df)
+
+    # Convert thousands of dollars to dollars
+    fgd_df.loc[:, fgd_df.columns.str.endswith("_1000_dollars")] = (
+        fgd_df.loc[:, fgd_df.columns.str.endswith("_1000_dollars")] * 1000
+    )
+    fgd_df.columns = fgd_df.columns.str.replace("_1000_dollars", "")  # Rename columns
+
+    # Convert SO2 test date column to datetime
+    # First, rescue a few troublesome datetimes that look like 3/08 or 01/08
+    troublesome_dates = fgd_df.so2_test_date.str.contains(
+        r"^[0-9]{1,2}\/[0-9]{1,2}$", regex=True, na=False
+    )
+    logger.info(
+        f"Rescuing troublesome dates: {fgd_df[troublesome_dates].so2_test_date.unique()}"
+    )
+    fgd_df.loc[troublesome_dates, "so2_test_date"] = fgd_df[
+        troublesome_dates
+    ].so2_test_date.str.pad(5, fillchar="0")
+    fgd_df.loc[troublesome_dates, "so2_test_date"] = pd.to_datetime(
+        fgd_df.loc[troublesome_dates, "so2_test_date"], format="%m/%y"
+    )
+
+    test_datetime = pd.to_datetime(
+        fgd_df.so2_test_date, format="mixed", errors="coerce", dayfirst=False
+    )
+    logger.info(
+        f"Dropping SO2 test dates that were not parseable: {fgd_df[(test_datetime.isnull())&(fgd_df.so2_test_date.notnull())].so2_test_date.unique()}"
+    )
+    fgd_df.loc[:, "so2_test_date"] = test_datetime
+
+    # Handle mixed boolean types in control flag column
+    fgd_df["fgd_control_flag"] = fgd_df["fgd_control_flag"].replace(
+        ["Y", "N", 1.0, 0.0], [True, False, True, False]
+    )
+
+    fgd_df["fgd_operational_status"] = fgd_df.fgd_operational_status.map(
+        pudl.helpers.label_map(
+            CODE_METADATA["core_eia__codes_operational_status"]["df"],
+            from_col="code",
+            to_col="label",  # OR operational_status? To discuss!
+            null_value=pd.NA,
+        )
+    )
+
+    # There are two remaining duplicates from plant_id_eia 6016, one row with cost data
+    # and one without. Keep the row with data.
+    dup_filt = (
+        (fgd_df.plant_id_eia == 6016)
+        & (fgd_df.so2_control_id_eia == 1)
+        & (fgd_df.report_year.isin([2018, 2019]))
+    )
+    # Drop the rows where every value other than the IDs and data maturity are null
+    fgd_df = fgd_df.loc[
+        ~((dup_filt) & (fgd_df.isna().sum(axis=1) >= len(fgd_df.columns) - 4))
+    ]
+
+    fgd_df = (
+        pudl.metadata.classes.Package.from_resource_ids()
+        .get_resource("core_eia__yearly_so2_control_equipment")
+        .encode(fgd_df)
+    )
```

Comment:
I think this is the step that I was talking about above that will run the encoder without your needing to directly reference the `CODE_METADATA` there could be a caveat that I'm not aware of though.